ui-tests/tools/com.liferay.ide.functional.modules.ext/com.liferay.ide.functional.modules.ext.wizard/com.liferay.ide.functional.modules.ext.wizard.validation/src/com/liferay/ide/functional/modules/ext/validation/tests/ValidationModulesExtFilesTests.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.ide.functional.modules.ext.validation.tests;
+
+import com.liferay.ide.ui.liferay.SwtbotBase;
+import com.liferay.ide.ui.liferay.support.project.ProjectSupport;
+import com.liferay.ide.ui.liferay.support.project.ProjectsSupport;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * @author Rui Wang
+ */
+public class ValidationModulesExtFilesTests extends SwtbotBase {
+
+	@Test
+	public void checkInfoInitialState() {
+		wizardAction.openNewLiferayWorkspaceWizard();
+
+		wizardAction.newLiferayWorkspace.prepareGradleWithIndexSources(project.getName());
+
+		wizardAction.finish();
+
+		jobAction.waitForNoRunningJobs();
+
+		wizardAction.openNewLiferayModulesExtWizard();
+
+		wizardAction.newModulesExt.prepare(projects.getName(0));
+
+		validationAction.assertEnabledTrue(wizardAction.newModulesExt.browseBtn());
+
+		wizardAction.newModulesExt.openSelectBrowseDialog();
+
+		dialogAction.prepareText("com.liferay:com.liferay.login.web");
+
+		dialogAction.confirm();
+
+		wizardAction.next();
+
+		validationAction.assertEquals(ADD_SOURCE_FILES_TO_OVERRIDE, wizardAction.getValidationMsg(1));
+
+		validationAction.assertEnabledTrue(wizardAction.newModulesExt.getAddOriginMoudleBtn());
+
+		validationAction.assertEnabledFalse(wizardAction.getNextBtn());
+
+		validationAction.assertEnabledTrue(wizardAction.getFinishBtn());
+
+		wizardAction.cancel();
+
+		viewAction.project.closeAndDeleteFromDisk(project.getName());
+	}
+
+	@Test
+	public void checkOpenExtFilesState() {
+		wizardAction.openFileMenuExtFilesWizard();
+
+		dialogAction.prepareText("Liferay Module Ext Files");
+
+		wizardAction.next();
+
+		validationAction.assertEquals(
+			UNABLE_TO_IDENTIFY_ORIGINAL_MODULE_SOURCE_IN_CURRENT_CONTEXT, wizardAction.getValidationMsg(1));
+
+		validationAction.assertEnabledTrue(wizardAction.getBackBtn());
```

Comment:
check "original Module" is not editable(false)
check Add files.. btn enable
click Add files.. btn
check dialog "unable to get source files in current context."